segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/ContainerRecoveryUtils.java

```diff
@@ -182,14 +184,14 @@ private static Map<Integer, Set<String>> getExistingSegments(Map<Integer, DebugS
         for (val containerEntry : containerMap.entrySet()) {
             Preconditions.checkNotNull(containerEntry.getValue());
             val tableExtension = containerEntry.getValue().getExtension(ContainerTableExtension.class);
-            val keyIterator = tableExtension.keyIterator(getMetadataSegmentName(
-                    containerEntry.getKey()), args).get(timeout.toMillis(), TimeUnit.MILLISECONDS);
+            val metadataSegmentName = isStorageMetadata ? getStorageMetadataSegmentName(containerEntry.getKey()) : getMetadataSegmentName(containerEntry.getKey());
+            val keyIterator = tableExtension.keyIterator(metadataSegmentName, args).get(timeout.toMillis(), TimeUnit.MILLISECONDS);
 
             // Store the segments in a set
             Set<String> metadataSegments = new HashSet<>();
             Futures.exceptionallyExpecting(keyIterator.forEachRemaining(k ->
                     metadataSegments.addAll(k.getEntries().stream()
-                            .map(entry -> entry.getKey().toString())
+                            .map(entry -> new String(entry.getKey().getCopy()))
```

Comment:
What is the motivation for this?

Reply:
converting segment names to human readable forms so that i can take actions on certain segments if needed.